### PR TITLE
[KOGITO-138] - Upgrade operator to use Go Mod instead of Dep

### PR DESCRIPTION
**Pull request by xiezhang7**
Jira: https://issues.jboss.org/browse/KOGITO-138

Upgrade the operator to use GO modules.
The operator currently uses dependency management tool "dep" which is supposed to be replaced with GO modules. 

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by xiezhang7** [comment]
Hi @ricardozanini @bmozaffa @tchughesiv , Could you please review this pull request? Thanks.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -9,26 +9,33 @@ import (
...
 )
 
 // Change below variables to serve metrics on different host or port.
 var (
-	metricsHost       = "0.0.0.0"
-	metricsPort int32 = 8383
-	log               = logger.GetLogger("cmd")
+	metricsHost               = "0.0.0.0"
+	metricsPort         int32 = 8383
+	operatorMetricsPort int32 = 8686
```
Comment (ricardozanini) [inline cmd/manager/main.go]: Very nice addition, but how the Operator Metrics is related to this migration?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	k8s.io/kube-state-metrics => k8s.io/kube-state-metrics v1.6.0
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.1.12
+	sigs.k8s.io/controller-tools => sigs.k8s.io/controller-tools v0.1.11-0.20190411181648-9d55346c2bde
+)
+
+replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.9.0
+
+replace (
+	github.com/openshift/api => github.com/openshift/api v3.9.1-0.20190814194116-a94e914914f4+incompatible
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20190813201236-5a5508328169
```
Comment (ricardozanini) [inline go.mod]: This release is from branch `release-4.3`? If the unit tests are working, I'm assuming is a version compatible with the fake client from Kubernetes.

File: hack/go-mod-env.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+setGoModEnv() {
+  pwdPath=$(pwd -P 2>/dev/null || env PWD= pwd)
+  goPath=$(go env GOPATH)
+  cd $goPath || exit
+  goPath=$(pwd -P 2>/dev/null || env PWD= pwd)
+  cd $pwdPath || exit
+  if [ "${pwdPath#"$goPath"}" != "${pwdPath}" ]; then
+#    echo "Inside GOPATH/src. Set GO111MODULE=on"
```
Comment (ricardozanini) [inline hack/go-mod-env.sh]: Leftovers from debug?

File: hack/go-mod.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+. ./hack/go-mod-env.sh
+
+echo Resetting vendor diectory
```
Comment (ricardozanini) [inline hack/go-mod.sh]: typo on directory

File: tools.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// +build tools
+
+package tools
+
+import (
+	_ "sigs.k8s.io/controller-tools/pkg/crd/generator"
+)
```
Comment (ricardozanini) [inline tools.go]: The reason of this file is to generate the `crd` once we update the `types` scheme?

Comment (xiezhang7) [reply tools.go]: This file is part of the scaffold generated by the operator-sdk when go-mod is enabled. It is used for tool dependencies when running the `operator-sdk generate` command with operator-sdk v0.9.0. The details are discussed in the pull requests: [go mod support for new operators 1](https://github.com/operator-framework/operator-sdk/pull/1001#issuecomment-464826207), [go mod support for new operators 2](https://github.com/operator-framework/operator-sdk/pull/1001#discussion_r251139696) and [tool dependencies](https://github.com/golang/go/issues/25922#issuecomment-402918061).

File: hack/go-mod.sh (hunk 4 above)
Comment (xiezhang7) [reply hack/go-mod.sh]: Fixed. Thanks.

File: hack/go-mod-env.sh (hunk 3 above)
Comment (xiezhang7) [reply hack/go-mod-env.sh]: Fixed. Thanks.

File: go.mod (hunk 2 above)
Comment (xiezhang7) [reply go.mod]: These versions are mapped from `release-4.3`. I specified `release-4.3` manually in the file and the versions are converted after executing `go mod vendor`.

File: cmd/manager/main.go (hunk 1 above)
Comment (xiezhang7) [reply cmd/manager/main.go]: Actually, these variables are generated by the operator-sdk. Operator-sdk v0.9.0 added a new feature: `Generating and serving info metrics about each custom resource. By default these metrics are exposed on port 8686`.  Before v0.9.0, metrics for the operator are exposed via port 8383. In addition to that, in v0.9.0 custom resource metrics are also exposed, via port 8686. In the meantime, there is a breaking change: `ExposeMetricsPort is removed and replaced with CreateMetricsService() function`, which supports exposing multiple ports for the metrics service. Please find the release notes [here](https://github.com/operator-framework/operator-sdk/releases/tag/v0.9.0). Thanks.

Comment (ricardozanini) [reply cmd/manager/main.go]: Hmm so I assume this should be implemented by other JIRA that aimed to upgrade to 0.9.0. As that JIRA was merged, I'll add a note about this one there.

Thanks for the detailed explanation, @xiezhang7

**Comment by xiezhang7** [comment]
Thanks, @ricardozanini @bmozaffa . I have squashed the commits.

**Pull request merged**